torch/lib/THD/base/init_methods/InitMethod.hpp

```diff
@@ -11,23 +11,56 @@ namespace thd {
 struct InitMethod {
   struct Config {
     struct MasterConfig {
-      rank_type world_size;
       int listen_socket;
       port_type listen_port;
     };
 
     struct WorkerConfig {
-      std::string address;
-      port_type port;
+      std::string master_addr;
+      port_type master_port;
     };
 
+    Config() {
+      rank = INT32_MAX - 1;
```

Comment:
If rank is an unsigned type you can do just `rank = -1`. It's well defined to be the largest representable value. You can also compare it to -1 later